packages/downloading-helpers/src/scripts/replay-assets.ts

```diff
@@ -23,45 +23,74 @@ interface AssetMetadata {
 const ASSETS_FOLDER_NAME = "assets";
 const ASSET_METADATA_FILE_NAME = "assets.json";
 
-export const fetchAsset: (path: string) => Promise<string> = async (path) => {
-  const logger = log.getLogger(METICULOUS_LOGGER_NAME);
+/**
+ * Downloads the given 'module' JS file as an MJS file (to force correct interpretation when loading the JS file).
+ *
+ * The associated source map will also be downloaded if present to sit alongside the main JS file and if the base snippets URL is a localhost URL.
+ */
+export const fetchAsset = async (path: string): Promise<string> => {
+  const snippetsBaseUrl = getSnippetsBaseUrl();
   const fetchUrl = new URL(path, getSnippetsBaseUrl()).href;
-  const assetFileName = `${basename(new URL(fetchUrl).pathname, ".js")}.cjs`;
+  const assetFileName = basename(new URL(fetchUrl).pathname);
+  const assetFileNameAsCjsFile = convertJsExtensionToCJS(assetFileName);
+
+  const jsFilePath = await fetchAndCacheFile(fetchUrl, assetFileNameAsCjsFile);
+  if (snippetsBaseUrl.includes("localhost")) {
+    await fetchAndCacheFile(`${fetchUrl}.map`, `${assetFileName}.map`);
```

Comment:
I think for this one you want?

```suggestion
    await fetchAndCacheFile(`${fetchUrl}.map`, `${assetFileNameAsCjsFile}.map`);
```

Reply:
Since the browser will load 'meticulous.bundle.cjs' and look for 'meticulous.bundle.cjs.map'

Reply:
It's actually not! I'm not exactly sure _why_ but both the Chrome debugger and the VS Code one seem to look in `.js.map` even if the file is a `.cjs` one.